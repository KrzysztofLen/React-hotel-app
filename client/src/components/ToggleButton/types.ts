export type Props = {
  onClick: (index: number) => void;
  isOpen: boolean;
  index: number;
  btnClass: string;
};
