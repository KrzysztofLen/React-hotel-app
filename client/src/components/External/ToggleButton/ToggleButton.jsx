import React from 'react';

class ToggleButton extends React.Component {
	constructor(props) {
		super(props);
		this.onToggleButton = this.onToggleButton.bind(this);
	}

	onToggleButton() {
		this.props.onClick(this.props.index);
	};

	render() {
		return <React.Fragment>
			<button className={this.props.btnClass}
			        onClick={this.onToggleButton}>{this.props.activeIndex ? "Less" : "More"}
			</button>
		</React.Fragment>
	}
}

export default ToggleButton;