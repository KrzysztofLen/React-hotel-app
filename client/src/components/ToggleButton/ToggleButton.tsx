import React from 'react';
import { Props } from './types';

const ToggleButton = (props: Props) => {
  const onToggleButton = () => {
    props.onClick(props.index);
  };

  return (
    <button className={props.btnClass} onClick={onToggleButton}>
      {props.isOpen ? 'Less' : 'More info'}
    </button>
  );
};

const MemoizedToggleButton = React.memo(ToggleButton);
export { MemoizedToggleButton as ToggleButton };
