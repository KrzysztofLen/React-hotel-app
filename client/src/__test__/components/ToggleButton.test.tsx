import React from 'react';
import { shallow } from 'enzyme';
import { ToggleButton } from '../../components/ToggleButton/ToggleButton';

describe('[COMPONENT] ToggleButton', () => {
  it('should render ToggleButton component correctly', () => {
    const wrapper = shallow(
      <ToggleButton index={1} isOpen={true} onClick={() => {}} btnClass={''} />,
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render inside test with More', () => {
    const wrapper = shallow(
      <ToggleButton
        index={1}
        isOpen={false}
        onClick={() => {}}
        btnClass={''}
      />,
    );
    expect(wrapper.text()).toEqual('More info');
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render inside test with Less', () => {
    const wrapper = shallow(
      <ToggleButton index={1} isOpen={true} onClick={() => {}} btnClass={''} />,
    );
    expect(wrapper.text()).toEqual('Less');
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('it call onToggleButton on button click', () => {
    const onToggleButton = jest.fn();
    const wrapper = shallow(
      <ToggleButton
        index={1}
        onClick={onToggleButton}
        isOpen={false}
        btnClass={''}
      />,
    );
    wrapper.find('button').simulate('click');
    expect(onToggleButton).toHaveBeenCalled();
  });
});
