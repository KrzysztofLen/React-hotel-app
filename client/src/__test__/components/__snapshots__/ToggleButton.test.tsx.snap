// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] ToggleButton should render ToggleButton component correctly 1`] = `
"<button className=\\"\\" onClick={[Function: onToggleButton]}>
  Less
</button>"
`;

exports[`[COMPONENT] ToggleButton should render inside test with Less 1`] = `
"<button className=\\"\\" onClick={[Function: onToggleButton]}>
  Less
</button>"
`;

exports[`[COMPONENT] ToggleButton should render inside test with More 1`] = `
"<button className=\\"\\" onClick={[Function: onToggleButton]}>
  More info
</button>"
`;
